js/src/main/scala/com/lynbrookrobotics/funkydashboard/Dataset.scala

```diff
@@ -2,44 +2,48 @@ package com.lynbrookrobotics.funkydashboard
 
 import slinky.core.facade.ReactElement
 import argonaut._, Argonaut._, ArgonautShapeless._
-
 import scala.collection.immutable.Queue
 
 object Dataset {
   def extract(definition: DatasetDefinition, sendData: String => Unit): Queue[TimedValue[String]] => ReactElement = {
     definition match {
-      case DatasetDefinition(_, "time-series-numeric") =>
+      case DatasetDefinition(_, tpe) if tpe.startsWith("time-series-numeric") =>
         values => TimeSeriesNumeric(
-          values.map(v => TimedValue(v.time, v.value.decodeOption[Double].get))
+          values.map(v => TimedValue(v.time, v.value.decodeOption[Double].get)),
+          tpe.drop(20)
+
         )
 
-      case DatasetDefinition(_, "time-multiple-dataset") =>
+      case DatasetDefinition(_, tpe) if tpe.startsWith("time-multiple-dataset") =>
         values => MultipleTimeSeriesNumeric(
-          values.map(v => TimedValue(v.time, v.value.decodeOption[List[Double]].get))
+          values.map(v => TimedValue(v.time, v.value.decodeOption[List[Double]].get)),
+          tpe.drop(22)
+
         )
 
-      case DatasetDefinition(_, "table") =>
+      case DatasetDefinition(_, tpe) if tpe.startsWith("table") =>
```

Comment:
I don't think this and image-stream onwards need to be changed, since they don't have units.